### PR TITLE
Add CyanoByte document validator using the JSON Schema

### LINKED ISSUE #7: Spec validator tool
There should be a tool that validates possible fields. This could be done as a custom linter, or convert the YAML to JSON as part of the linter and use [JSON Schema](https://json-schema.org/).

Once this happens, we should add CI tests for it and keep it up to date as we make spec changes.

### PR DESCRIPTION
**Pull request by chrisfrederickson**
This adds a CyanoByte document validator using the JSON Schema from #27. It also adds a version of the example from `docs/cyanobyte.md` to `peripherals` for testing. To verify the functionality, run:

```
python3 src/validator.py -i peripherals/MCP9808.yaml
```

The validation fails on `peripherals/Mcp4725.yaml` as there are a few features in that spec that have not been added to the JSON Schema validator file. Fixes #7.

### CODE REVIEW

File: src/validator.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    Args:
+        input_files: A list of CyanoByte documents to validate.
+    """
+    # Load the JSON Schema file
+    with open("spec/cyanobyte.schema.json", "r") as schema_json:
+        schema = json.load(schema_json)
+
+    # Validate each document against the schema
+    for input_file in input_files:
```
Comment (Fleker) [inline src/validator.py]: Should this be indented one block, so that schema will be defined?

Comment (chrisfrederickson) [reply src/validator.py]: Most scoping in python is not local to loops / with statements. This runs fine. Can't find a concise explanation of scoping rules in python, but this is the closest reference I could find: https://mail.python.org/pipermail/python-ideas/2008-October/002109.html

**Comment by chrisfrederickson** [comment]
Working on adding the unit tests.

File: src/validator.py (hunk 1 above)
Comment (Fleker) [reply src/validator.py]: Sure, but AFAIK if the file open fails, schema won't be defined, causing this for loop to fail.

Comment (chrisfrederickson) [reply src/validator.py]: Just double checked, if the file open fails, it will still raise a `FileNotFoundError` or `IOError` exception.

**Comment by chrisfrederickson** [comment]
@Fleker This is more or less done, but #43 needs to be merged first so as to appease CloudBuild.

File: test/test_validator.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import unittest
```
Comment (Fleker) [inline test/test_validator.py]: Can you add this script to the `cloudbuild.yaml` file?

**Comment by Fleker** [comment]
You'll need to rebase first

**Comment by Fleker** [comment]
LGTM

**Pull request merged**